src/Migration.php

```diff
@@ -181,7 +197,38 @@ public function setModel(Model $m)
                 continue;
             }
 
-            $this->field($field->actual ?: $field->short_name, ['type' => $field->type]);  // todo add more options here
+            // get field type from field
+            $type = $field->type;
+
+            // if the field is a hasOne relation
+            // Don't have the right FieldType
+            // FieldType is stored in the reference field
+            if ($field->reference instanceof HasOne) {
+
+                // @TODO if this can be done better?
```

Comment:
i leave a TODO, probably can done better.
I need to get the field type of the Referenced hasOne.
To find it : 
 - i create the referenced model stored in $field->reference->model, using the actual persistence/connection
 - i call $refModel->get($field->reference->their_field) but $their_field is protected
 - i think $their_field must remain protected
 - for this specific case, i think the best option is to use Reflection
 - if $their_field is null use $id_field of the referenced model
 - if try to get the type of this field and if is null i return integer

I use reflection because migration is not a normal "production" and performance are not so important here.